lib/src/tests/mod.rs

```diff
@@ -0,0 +1,225 @@
+        Stmt::Return(expr) => Return(expr_tast_to_ast(expr)),
+        Stmt::While(cond, body) => While(expr_tast_to_ast(cond), Box::new(stmt_tast_to_ast(body))),
+        Stmt::If(cond, body, elze) => If(
+            expr_tast_to_ast(cond),
+            Box::new(stmt_tast_to_ast(body)),
+            match elze {
+                Some(x) => Some(Box::new(stmt_tast_to_ast(x))),
+                None => None,
+            },
+        ),
+        Stmt::StmtExprStmt(stmt_expr) => StmtExprStmt(stmt_expr_tast_to_ast(stmt_expr)),
+        default => stmt.clone(),
+    }
+}
+
+fn stmt_expr_tast_to_ast(stmt_expr: &StmtExpr) -> StmtExpr {
+    match stmt_expr {
+        StmtExpr::Assign(var, val) => Assign(var.clone(), expr_tast_to_ast(val)),
+        StmtExpr::New(typ, params) => New(
+            typ.clone(),
+            params.iter().map(|x| expr_tast_to_ast(x)).collect(),
+        ),
+        StmtExpr::MethodCall(obj, method, params) => MethodCall(
+            expr_tast_to_ast(obj),
+            method.clone(),
+            params.iter().map(|x| expr_tast_to_ast(x)).collect(),
+        ),
+        StmtExpr::TypedStmtExpr(x, typ) => stmt_expr_tast_to_ast(x),
+    }
+}
+
+fn expr_tast_to_ast(expr: &Expr) -> Expr {
+    match expr {
+        Expr::InstVar(x, s) => InstVar(Box::new(expr_tast_to_ast(x)), s.clone()),
+        Expr::Unary(s, x) => Unary(s.clone(), Box::new(expr_tast_to_ast(x))),
+        Expr::Binary(op, l, r) => Binary(
+            op.clone(),
+            Box::new(expr_tast_to_ast(l)),
+            Box::new(expr_tast_to_ast(r)),
+        ),
+        Expr::StmtExprExpr(x) => StmtExprExpr(Box::new(stmt_expr_tast_to_ast(x))),
+        Expr::TypedExpr(x, t) => expr_tast_to_ast(x),
+        default => expr.clone(),
+    }
+}
+
+fn tast_to_ast(class: &Class) -> Class {
+    Class {
+        name: class.name.clone(),
+        fields: class.fields.clone(),
+        methods: class
+            .methods
+            .clone()
+            .into_iter()
+            .map(|method| MethodDecl {
+                ret_type: method.ret_type.clone(),
+                name: method.name.clone(),
+                params: method.params.clone(),
+                body: stmt_tast_to_ast(&method.body),
+            })
+            .collect(),
+    }
+}
+
+fn create_test_file(ast: &Class, tast: Option<&Class>, name: &str) {
+    let file_path = format!("tests/{name}.java");
+    let gen_file_path = format!("tests/{name}-gen.java");
+    let class_file_path = format!("tests/{name}.class");
+
+    // Generate Java Code from AST and write to file
+    let class_code = class_to_java(ast);
+    let mut file =
+        File::create(gen_file_path.clone()).expect("File for generated code couldn't be created");
+    file.write(class_code.as_bytes())
+        .expect("Couldn't write generate java code");
+
+    // TODO: Check that generated java code and original java code are equivalent to javac
```

Comment:
Though it should be said that javac probably generates much better optimized Java Code with more special instructions. Would probably interesting whether we can tune that a bit by idk using older javac versions?

Reply:
In that comment I meant the code that I already generate via the AST. That test would basically just check that I didn't fuck up when writing the AST from hand.

How we can test the bytecode that you generate is a bit of a harder problem, that's true.

Reply:
Oh yes that would be really nice